### PR TITLE
Allow a notification spec run only once per day

### PR DESCRIPTION
**Pull request by we-ai**
Updates in this PR:
- Implement requests in [issue #930](https://github.com/episphere/connect/issues/930)
- Print count of updated participant documents before finishing `updateParticipantData`, for troubleshooting in future

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -2065,7 +2065,30 @@ const getScheduledNotifications = async (scheduleAt) => {
...
+ */
+const getNotificationSpecsByScheduleOncePerDay = async (scheduleAt) => {
+  const currTime = new Date().toISOString();
+  const batch = db.batch();
+  const snapshot = await db.collection("notificationSpecifications").where("scheduleAt", "==", scheduleAt).get();
+  let notificationSpecArray = [];
+  for (const doc of snapshot.docs) {
+    const docData = doc.data();
+    const lastRunDate = docData.lastRunTime?.split("T")[0] ?? "";
+    if (!docData.isDraft && docData.id && currTime.split("T")[0] !== lastRunDate) {
```
Comment (anthonypetersen) [inline utils/firestore.js:2081]: very nice

File: utils/sites.js [hunk 2]
```diff
@@ -172,7 +172,7 @@ const siteNotificationsHandler = async (Connect_ID, concept, siteCode, obj) => {
 }
 
 const updateParticipantData = async (req, res, authObj) => {
-    const { getParticipantData, updateParticipantData, writeCancerOccurrences } = require('./firestore');
+    const { getParticipantData, updateParticipantData: updateParticipantDataFirestore, writeCancerOccurrences } = require('./firestore');
```
Comment (anthonypetersen) [inline utils/sites.js:175]: should this be a comma?

**Review by anthonypetersen — APPROVED** [review]
just one comment - besides that looks very good, thanks for hopping on this so quickly

File: utils/sites.js (hunk 2 above)
Comment (anthonypetersen) [reply utils/sites.js:175]: interesting, haven't seen that syntax before

File: utils/firestore.js (hunk 1 above)
Comment (we-ai) [reply utils/firestore.js:2081]: Thanks. I just realize that we should use US east timezone instead of UTC time to check a date. Doing updates now.

**Review by anthonypetersen — APPROVED** [review]

**Review by bransteitterbr — APPROVED** [review]

**Pull request merged**